.github/workflows/go-mod-tidy.yml

```diff
@@ -0,0 +1,28 @@
+name: Go Mod Tidy Check
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+  release:
+    types: [published]
+
+env:
+  GO_VERSION: 1.18
```

Comment:
We really need to have some global var for this. Last time I made an update, I had to update this field in like ~7 places

Reply:
Ideally we should combine all our Go CI into a single file. Filing an issue